scripts/backup.sh

```diff
@@ -66,84 +81,125 @@ ROLLING="$BASEDIR/rolling/backup_$DOW.tar.gz"
 [ -d ./.tmp/backup ] || mkdir -p ./.tmp/backup
 [ -d ./.tmp/databases_backup ] || mkdir -p ./.tmp/databases_backup
 
-touch $LOGFILE
-echo ""  > $LOGFILE
-echo "### IOTstack backup generator log ###" >> $LOGFILE
-echo "Started At: $(date +"%Y-%m-%dT%H-%M-%S")" >> $LOGFILE
-echo "Current Directory: $(pwd)" >> $LOGFILE
-echo "Backup Type: $BACKUPTYPE" >> $LOGFILE
+#------------------------------------------------------------------------------
+# echo pass params and log to file LOGFILE
+# depend to global var VERBOSE
+# Example:
+# doLog "Something"
+#------------------------------------------------------------------------------
+doLog(){
+  # also we can log every command with timestamp
+  MSG=''
+  # Check if string is empty print empty line without timestamp
+  if [ -z "$*" ]
+  then
+    MSG=$(echo "$*")
+  else
+    MSG=$(echo "$(date '+%Y.%m.%dT%H:%M:%S') - ""$*")
+  fi
+  # with colors code it's looks very ugly
+  # taken from: https://stackoverflow.com/a/51141872/660753
+  MSG=$(echo "$MSG" | sed 's/\x1B\[[0-9;]\{1,\}[A-Za-z]//g')
+
+  # write to console if VERBOSE is set
+  if [ $VERBOSE ]
+  then
+    # also on the screen lines not looking cool, remove \r symbol
+    echo "$MSG" | sed 's/\r/\n/g'
+  fi
+
+  # write to LOGFILE
+  echo "$MSG" >> "$LOGFILE"
+}
+#eof func doLog
+
+touch "$LOGFILE"
+doLog ""
+doLog "### IOTstack backup generator log ###"
+doLog "Started At: $(date +"%Y-%m-%dT%H-%M-%S")"
+doLog "Current Directory: $(pwd)"
+doLog "Backup Type: $BACKUPTYPE"
 
 if [[ "$BACKUPTYPE" -eq "1" || "$BACKUPTYPE" -eq "3" ]]; then
-  echo "Backup File: $BACKUPFILE" >> $LOGFILE
+  doLog "Backup File: $BACKUPFILE"
 fi
 
 if [[ "$BACKUPTYPE" -eq "2" || "$BACKUPTYPE" -eq "3" ]]; then
-  echo "Rolling File: $ROLLING" >> $LOGFILE
+  doLog "Rolling File: $ROLLING"
 fi
 
-echo "" >> $BACKUPLIST
+echo "" >> "$BACKUPLIST"
 
-echo "" >> $LOGFILE
-echo "Executing prebackup scripts" >> $LOGFILE
-bash ./scripts/backup_restore/pre_backup_complete.sh >> $LOGFILE 2>&1
+doLog ""
+doLog "Executing pre-backup scripts"
+# Output to log file
+docker-compose logs --no-color --tail="50" >> "$LOGFILE"
+doLog "$(bash ./scripts/backup_restore/pre_backup_complete.sh)"
 
-echo "./services/" >> $BACKUPLIST
-echo "./volumes/" >> $BACKUPLIST
-[ -f "./docker-compose.yml" ] && echo "./docker-compose.yml" >> $BACKUPLIST
-[ -f "./compose-override.yml" ] && echo "./compose-override.yml" >> $BACKUPLIST
-[ -f "./extra" ] && echo "./extra" >> $BACKUPLIST
-[ -f "./.tmp/databases_backup" ] && echo "./.tmp/databases_backup" >> $BACKUPLIST
-[ -f "./postbuild.sh" ] && echo "./postbuild.sh" >> $BACKUPLIST
-[ -f "./post_backup.sh" ] && echo "./post_backup.sh" >> $BACKUPLIST
-[ -f "./pre_backup.sh" ] && echo "./pre_backup.sh" >> $BACKUPLIST
+echo "./services/" >> "$BACKUPLIST"
+echo "./volumes/" >> "$BACKUPLIST"
+[ -f "./docker-compose.yml" ] && echo "./docker-compose.yml" >> "$BACKUPLIST"
+[ -f "./compose-override.yml" ] && echo "./compose-override.yml" >> "$BACKUPLIST"
```

Comment:
You may need to add `docker-compose-override.yml` here too.

Reply:
Was added